### PR TITLE
Fabric 2.x chaincode lifecycle update

### PR DESCRIPTION
**Pull request by himanshulalarya**
**Changelog**
- Updated version changes and chaincode lifecycle for fabric 2.x
- Added new chart for chaincode approve
- Updated Fabric Binaries

 

**Reviewed by**
@lakshyakumar @jagpreetsinghsasan 

 

**Linked issue**
#966 #963

### CODE REVIEW

File: platforms/hyperledger-fabric/configuration/roles/create/chaincode/commit/tasks/write.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Copy organization level certificates for orgs {{ endorser.name }}
+  shell: |
+    vault write secret/crypto/peerOrganizations/{{ namespace }}/endorsers/{{ endorser.name }}/users/admin/msp cacerts="$(cat ./build/crypto-config/peerOrganizations/{{ endorser.name }}-net/users/Admin@{{ endorser.name }}-net/msp/cacerts/ca-{{ endorser.name }}-net-7054.pem)"
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
```
Comment (suvajit-sarkar) [inline platforms/hyperledger-fabric/configuration/roles/create/chaincode/commit/tasks/write.yaml:7]: new line please

File: platforms/shared/configuration/roles/helm_lint/vars/main.yaml [hunk 2]
```diff
@@ -39,4 +39,6 @@ charts:
   node_registration: node-initial-registration
   cli: fabric_cli
   commit_chaincode_job: commit_chaincode 
+  approve_chaincode_job: approve_chaincode
   node: node
+
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/roles/helm_lint/vars/main.yaml:44]: additional newline might be the reason of linting error in molecule, run the molecule test and fix the errors

File: platforms/hyperledger-fabric/configuration/roles/helm_component/templates/cli.tpl [hunk 3]
```diff
@@ -33,6 +33,6 @@ spec:
       name: {{ peer.name }}
       localmspid: {{ org.name | lower}}MSP
       tlsstatus: true
-      address: {{ peer.gossippeeraddress }}
+      address: {{ peer.name }}.{{ component_ns }}.{{ external_url_suffix }}:8443
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/helm_component/templates/cli.tpl:36]: refer comments in PR #1040

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-fabric/charts/create_channel/templates/create_channel.yaml [hunk 4]
```diff
@@ -148,9 +148,19 @@ spec:
             echo "Creating Channel ${CHANNEL_NAME}"
             if [ "$tls_status" = "true" ]
             then
-              peer channel create -o ${ORDERER_URL} -c ${CHANNEL_NAME} -f channel.tx --tls ${CORE_PEER_TLS_ENABLED} --cafile ${ORDERER_CA}
+              if [ $.Values.metadata.network.version == '1.4.4']
```
Comment (sownak) [inline platforms/hyperledger-fabric/charts/create_channel/templates/create_channel.yaml:151]: Why 1.4.4 and not 1.4.8?

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
If there are changes done to network.yaml, then change the sample network.yaml as well

**Comment by himanshulalarya** [comment]
Creating a new pr for this issue as some checks were not signed

**Pull request closed without merging**